modules/dhp_connected_devices/app/services/fitbit_service.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require 'fitbit/client'
+
+class FitbitService
+  def get_connection_status(data)
+    @fitbit_client = data[:fitbit_api]
+    data[:callback_params][:code] && exchange_code_for_token(data[:callback_params][:code]) ? 'success' : 'error'
+  end
+
+  private
+
+  def exchange_code_for_token(code)
+    response = @fitbit_client.get_token(code)
```

Comment:
This seems like an issue. We call `@fitbit_client` in this method because we assume that it was initialized in the previous method. This introduces a dependency between those 2 methods. So we must call `get_connection_status` before we call `exchange_code_for_token`. Otherwise it will error because `@fitbit_client` is not initialized.